### DISCUSSION TITLE
geometry to geometry distance？

### DISCUSSION THREAD

**Discussion by remotesc2** [Turfjs/turf#2331] [Q&A]
i`m install turf@6.5

I have a question. How can I calculate geometry to geometry shortest distance?

1. point to polygon
2. polyline to polygon
3. polygon to polygon
4. ...

i only found “turf-point-to-line-distance”

example like: https://developers.arcgis.com/javascript/3/jsapi/esri.geometry.geometryengine-amd.html#distance
![image](https://user-images.githubusercontent.com/9023036/189162927-eb0c14c9-cafc-486e-afbd-b9f1e34abf4e.png)

**Comment by twelch** [comment]
I don't think there is a robust/generic distance method like you are looking for that takes any geometry.  A brute force method could be to iterate through the coordinates in both geometries (using for example turf.coordEach()) calculating the shortest distance between each possible pair of coordinates to find the shortest distance, using turf.distance(p1, p2) or the possibly faster cheap-ruler library - https://github.com/mapbox/cheap-ruler

**Reply by remotesc2** [reply]
thank you for answer my question.
I used to think like this, use “forEach”, then, i have to face some problems. The first one is "corss", and I don't know if there are any other scenes. 
If there is only one special scene, i think i can use "forEach" instead “geometry to geometry”
